_posts/2020-12-01-infrastructure-migration-finished.md

```diff
@@ -0,0 +1,275 @@
+  We supplied VMs with the current version, so people could experiment with it.
+  I never heard back from ANY of them.
+  Like .. nothing after the initial offer and talk.
+
+There are many more examples, but I am sure you are understanding by now where I am coming from.
+And do not get me wrong: all of those people who offered to help, they did it with the best intentions, and I thank them for reaching out.
+But creating software and maintaining infrastructure is hard and difficult.
+It is not something you do on a Sunday afternoon and be done with it.
+It takes serious time and commitment.
+And myself running towards my 40s, time is not something that is ample anymore.
+It was when were were in our 20s, and the university allowed many many many many hours of doing this crazy things.
+
+So, in 2018, we decided this had to change.
+A few drastic changes were made, and many more would follow in the next two years.
+
+The big thing to solve: so far we have been using what is called unmanaged hosting.
+This means we get access to a server, and after that we have to do everything ourself.
+This is very cheap, and the main reason we used to do this; but it also means you have to put in the hours to maintain it.
+As pointed out above .. this was not happening.
+So we wanted to switch to managed hosting.
+This is a lot more expensive, but also would mean we didn't have to put in the work.
+With managed hosting, the provider is taking care of many things:
+
+- OS updates
+- Security patching
+- If you use a [SaaS (Software-as-a-Service)](https://en.wikipedia.org/wiki/Software_as_a_service), keeping the application up-to-date
+
+In 2018, one of the easiest gains was migrating from SubVersion to git, and using the [GitHub](https://github.com) to host that.
+This means we didn't have to maintain that part of the infrastructure.
+We also migrated our bug-tracker to GitHub.
+I mean, if you are going to use GitHub, do it properly or go home.
+
+Next, what about all the other services we run?
+We tried out several things, but in the end, I decided it was best to use cloud-solutions.
+We asked around every cloud provider if they offer anything for Open Source projects to soften the blow of the increase cost.
+I had many "fun" conversations .. luckily we found AWS answering the phone, with their [AWS Open Source credits program](https://aws.amazon.com/blogs/opensource/aws-promotional-credits-open-source-projects/).
+They funded 2020 for us, allowing me to experiment with all their services.
+This really really helped out, and so we got started in migrating to AWS.
+
+The initial work for this started in January 2019, although we were aiming for [DigitalOcean](https://www.digitalocean.com/)'s [Kubernetes](https://www.digitalocean.com/products/kubernetes/) solution at that time.
+For various of reasons this did not work out (mainly their lack of IPv6 for Spaces was a bit of a show-stopper; but we had various of other issues).
```

Comment:
\+ "at the time" ? (Assuming it has actually been resolved now)

Reply:
It has not :P